platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/cli/BuildActionsFactory.java

```diff
@@ -243,21 +239,17 @@ private Runnable runBuildInSingleUseDaemon(StartParameterInternal startParameter
         //end of workaround.
 
         // Create a client that will not match any existing daemons, so it will always start a new one
-        ServiceRegistry clientSharedServices = createGlobalClientServices(true);
+        ServiceRegistry clientSharedServices = createGlobalClientServices();
         ServiceRegistry clientServices = clientSharedServices.get(DaemonClientFactory.class).createSingleUseDaemonClientServices(clientSharedServices, daemonParameters, requestContext, System.in);
         DaemonClient client = clientServices.get(DaemonClient.class);
         return runBuildAndCloseServices(startParameter, daemonParameters, client, clientSharedServices, clientServices);
     }
 
-    private ServiceRegistry createGlobalClientServices(boolean usingDaemon) {
+    private ServiceRegistry createGlobalClientServices() {
         ServiceRegistryBuilder builder = ServiceRegistryBuilder.builder()
             .displayName("Daemon client global services")
             .parent(NativeServices.getInstance());
-        if (usingDaemon) {
```

Comment:
This conditional was not necessary. The "basicServices" could be used in all cases.